src/point3d.py

```diff
@@ -20,7 +20,7 @@ def distance_from_origin(self) -> float:
         """
         return self.distance_from_point(Point3D(0, 0, 0))
 
-    def distance_from_point(self, other) -> float:
+    def distance_from_point(self, other: "Point3D") -> float:
```

Comment:
Self type hinting is usually done via an import.

```python
from __future__ import annotations

class A:
    def f(a: A):
        pass
```

I'm not sure if `"Point3D"` is the same, but I'd guess not.

Reply:
This was not done due to the import and overhead. It appears that a simple, and workable alternative is the string literal. 

Things fortunately seem smart enough to connect the dots. 
![image](https://user-images.githubusercontent.com/1328048/172450294-d2197c9a-cab3-4c2a-8133-9a73fd8f6c2e.png)

Reply:
[Also, reference to the PRP on it.](https://peps.python.org/pep-0484/#forward-references)